### DISCUSSION TITLE
localcluster denied

### DISCUSSION THREAD

**Discussion by stadox** [rackslab/Slurm-web#372] [Q&A] [answered]
Hello,

I would like to ask you what can be the reason why I'm not able to access my localcluster.
I have attached document with some pictures and overview of ini files.

Thank you very much for kind help and advice.

Regards,
Stanislav
[doc.docx](https://github.com/user-attachments/files/17585241/doc.docx)

![1](https://github.com/user-attachments/assets/8998b907-2abb-4cc1-b824-f0f8ad42b91f)
![2](https://github.com/user-attachments/assets/99629b50-92f2-4731-9b95-cd97289283b2)

**Comment by rezib** [comment]
What is the content of your `/etc/slurm-web/gateway.ini` and `/etc/slurm-web/agent.ini`? Take care of redacting passwords and confidential data.

**Reply by stadox** [reply]
vim /etc/slurm-web/gateway.ini

```ini
[ui]
host=http://localhost:8090

[agents]
url=http://localhost/agent

[service]
interface=localhost

[jwt]

key=/var/lib/slurm-web/jwt.key

duration=1

algorithm=HS256

audience=slurm-web
```
********************************************************

vim /etc/slurm-web/agent.ini
```ini
[service]
cluster=localcluster

[cache]
enabled=yes
```
********************************************************

vim /usr/share/slurm-web/wsgi/gateway/slurm-web-gateway.ini
```ini
[authentication]
auth_type = simple

[web]
enable_login = true
session_timeout = 3600
cluster_name = localcluster

[uwsgi]
env = LANG=en_US.UTF-8
plugin = python3

wsgi-file = /usr/share/slurm-web/wsgi/gateway/slurm-web-gateway.py

master = true
processes = 5

socket = /run/slurm-web-gateway/uwsgi.sock

chmod-socket = 666

buffer-size = 32768
vacuum = true

die-on-term = true

ignore-sigpipe=true
ignore-write-errors=true
disable-write-exception=true

[database]
host=localhost
user=slurm
port=3306
password=***********
database=slurm_acct_db
```
********************************************************
vim /usr/share/slurm-web/wsgi/agent/slurm-web-agent.ini
```ini
[uwsgi]
env = LANG=en_US.UTF-8
plugin = python3

wsgi-file = /usr/share/slurm-web/wsgi/agent/slurm-web-agent.py

master = true
processes = 5

socket = /run/slurm-web-agent/uwsgi.sock

chmod-socket = 666

buffer-size = 32768
vacuum = true

die-on-term = true
```

**Reply by rezib** [reply]
What is the purpose of `[authentication]`, `[web]` and `[database]` sections in `/usr/share/slurm-web/wsgi/gateway/slurm-web-gateway.ini`? These sections are not defined in [the file provided by Slurm-web](https://github.com/rackslab/Slurm-web/blob/main/lib/wsgi/gateway/slurm-web-gateway.ini). I think you should restore original file.

Also section `[jwt]` is useless in `/etc/slurm-web/gateway.ini` as you set default values.

The denied error in the web UI indicates anonymous user account does not have permission in the policy. Have you modified the authorization policy?

**Reply by stadox** [reply]
I have found this cases of configuration on in some use case related to /usr/share/slurm-web/wsgi/gateway/slurm-web-gateway.ini

Therefore I have restored ini files to the original as well as policy.ini 

**slurm-web-agent.ini**

[uwsgi]
env = LANG=en_US.UTF-8
plugin = python3

wsgi-file = /usr/share/slurm-web/wsgi/agent/slurm-web-agent.py

master = true
processes = 5

socket = /run/slurm-web-agent/uwsgi.sock

chmod-socket = 666

buffer-size = 32768
vacuum = true

die-on-term = true

***************************

**slurm-web-gateway.ini**

[uwsgi]
env = LANG=en_US.UTF-8
plugin = python3

wsgi-file = /usr/share/slurm-web/wsgi/gateway/slurm-web-gateway.py

master = true
processes = 5

socket = /run/slurm-web-gateway/uwsgi.sock

chmod-socket = 666

buffer-size = 32768
vacuum = true

die-on-term = true

ignore-sigpipe=true
ignore-write-errors=true
disable-write-exception=true

***************************

policy.ini

[roles]
anonymous

user=ALL

[anonymous]
actions=view-stats,view-jobs,view-nodes

[user]
actions=view-stats,view-jobs,view-nodes,view-partitions,view-qos,view-accounts,view-reservations

and now I'm seeing this kind of new issue:

![error](https://github.com/user-attachments/assets/0709c20d-9781-4eb9-9e56-386c4ae0c4e1)

![3](https://github.com/user-attachments/assets/fea824a6-f81a-42be-b08f-26dfb91f38a5)

**journalctl -u slurm-web-agent-uwsgi.service**

![4](https://github.com/user-attachments/assets/237935f4-5a7c-4202-84be-15a9f1bd0b31)

**slurmrestd.service**

[Unit]
Description=Slurm REST daemon
After=network-online.target remote-fs.target slurmctld.service
Wants=network-online.target
Requires=slurmctld.service
ConditionPathExists=/etc/slurm/slurm.conf

[Service]
Type=simple
EnvironmentFile=-/etc/sysconfig/slurmrestd
EnvironmentFile=-/etc/default/slurmrestd
ExecStart=/usr/sbin/slurmrestd $SLURMRESTD_OPTIONS
Environment=SLURM_JWT=daemon
Environment=SLURMRESTD_LISTEN=:6820
ExecReload=/bin/kill -HUP $MAINPID
User=slurmrestd
Group=slurmrestd
Restart=on-failure
PIDFile=/run/slurm/slurmrestd.pid

[Install]
WantedBy=multi-user.target



Thank you very much!

**Reply by rezib** [reply, marked as answer]
It seems you are impacted by https://github.com/rackslab/Slurm-web/issues/350. What version of python requests are you using? If you are using requests 2.32+, a workaround is to downgrade to requests 2.31, if possible.

The fix is included in upcoming release of Slurm-web 4.0.0, and it will probably be included in the next maintenance release in the 3.x branch.

**Reply by stadox** [reply]
Hello,

Thank you very much for response!

OS ROCKY LINUX and python-request as below shown:

![python](https://github.com/user-attachments/assets/63a3edfd-d515-4824-8450-98d7195d094b)

Thank you

Kind regards,
Stanislav

**Reply by rezib** [reply]
Then there is probably another bug with this version of python Requests on this OS… I just opened issue #373 to dig deeper.
